text/0011-utxo-model/0011-utxo-model.md

```diff
@@ -0,0 +1,194 @@
+
+  Instead of having to walk through multiple transaction to eventually find the spent balance in the tangle, we can now
+instantly load the corresponding funds by accessing the consumed outputs. A transaction that tries to spend non-existing
+funds would never become solid.
+
+- Accordingly, a transaction is performing a double spend when there is already another transaction consuming the same
+output.
+
+# Drawbacks
+
+- A UTXO-based ledger is much more complex in the implementation than just a dictionary of balances.
+- A UTXO-based ledger stores more information and therefore requires "potentially" more disk space.
+  
+  However, if
+sweeping becomes a central part of the protocol with all wallets supporting it, then this solution will most
+probably not perform much worse than an account-based ledger.
+
+  Considering how we store ledger state today, with several *checkpoints* where we *cache the ledger state* to reduce
+  possible tangle walking time, it might also be possible that the resource consumption drops because we are no longer
+  required to have these additional checkpoints with their corresponding ledger state (in memory and on disk).
+
+  In general, it does not introduce a new attack vector because it is already today possible to spread out small
+  balances onto many addresses, which is the equivalent of spreading a lot of outputs on a single address.
+
+# Rationale and alternatives
+
+There are currently only two options for a ledger state, that are known - the UTXO model, and the account model:
+
+- The account-model does not work very well with "parallelism" (i.e. multiple people sending funds to the same
+address). We can never be sure, that our transaction has seen all the funds and the only way to make sure, that this is
+the case is to use the tangle structure to define a *scope* for a spend. This however is very slow, because the only way
+to use the tangle structure is to walk around in the tangle. In addition, it adds additional constraints to the tip
+selection as it requires to have the correct funding transactions in the past cone of a spending transaction.
+
+- Considering the "parallel" nature of the tangle, it seems like UTXO is the only reasonable choice. Using addresses as
+the additional identifier for outputs, allows us to bundle multiple colored balances in a single output, which leads to
+a smaller fragmentation of the funds as if we would give every color a separate output.
+
+If we decide to not go with this approach, then we will essentially limit ourselves to just a few hundred TPS because of
+the massive overhead of checking incoming transactions. In addition, the conflict detection will be much harder.
+
+# Unresolved questions
+
+- We need to think about good unlock opcodes and options.
+- We should define a PR that specifies the "branch based ledger state" as this is very closely related to the
```

Comment:
a few words or a link on "branch based ledger state"?

Reply:
I think that concepts like reality/branch based ledgers is many layers away from plain UTXO transactions and relies on much more concepts. So, this should not be mentioned as an unresolved question for plain UTXO at all.